### DISCUSSION TITLE
Self-join removal deprecated

### DISCUSSION THREAD

**Discussion by marioscrock** [morph-kgc/morph-kgc#273] [General]
@arenas-guerrero-julian could you please explain why the self-join removal procedure is marked as to be deprecated?

https://github.com/morph-kgc/morph-kgc/blob/3be33a72c86166984f64ca2f890221669f97465f/src/morph_kgc/mapping/mapping_parser.py#L806-L807

**Comment by arenas-guerrero-julian** [comment]
The join can be removed when the reference is unique (see #209 for an example of issue that may arise). Because redundant self joins can always be removed manually if performance is not good, an option is to just remove this feature from Morph-KGC. Other possibility is no have an option in the config to let the user decide whether self join elimination should be applied.

**Reply by marioscrock** [reply]
Thanks a lot for pointing me to the correct issue! I was precisely looking for a similar example.
